modules/gapi/samples/text_detection.cpp

```diff
@@ -0,0 +1,705 @@
+        }
+    }
+
+    static int topLeftPointIdx(const std::vector<cv::Point2f> &points) {
+        // NOTE: Taken from the OMZ text detection sample almost as-is
+        cv::Point2f most_left(std::numeric_limits<float>::max(),
+                              std::numeric_limits<float>::max());
+        cv::Point2f almost_most_left(std::numeric_limits<float>::max(),
+                                     std::numeric_limits<float>::max());
+        int most_left_idx = -1;
+        int almost_most_left_idx = -1;
+
+        for (size_t i = 0; i < points.size() ; i++) {
+            if (most_left.x > points[i].x) {
+                if (most_left.x < std::numeric_limits<float>::max()) {
+                    almost_most_left = most_left;
+                    almost_most_left_idx = most_left_idx;
+                }
+                most_left = points[i];
+                most_left_idx = static_cast<int>(i);
+            }
+            if (almost_most_left.x > points[i].x && points[i] != most_left) {
+                almost_most_left = points[i];
+                almost_most_left_idx = static_cast<int>(i);
+            }
+        }
+
+        if (almost_most_left.y < most_left.y) {
+            most_left = almost_most_left;
+            most_left_idx = almost_most_left_idx;
+        }
+        return most_left_idx;
+    }
+
+}; // GAPI_OCV_KERNEL(CropLabels)
+
+} // anonymous namespace
+} // namespace custom
+
+namespace vis {
+namespace {
+
+void drawRotatedRect(cv::Mat &m, const cv::RotatedRect &rc) {
+    std::vector<cv::Point2f> tmp_points(5);
+    rc.points(tmp_points.data());
+    tmp_points[4] = tmp_points[0];
+    auto prev = tmp_points.begin(), it = prev+1;
+    for (; it != tmp_points.end(); ++it) {
+        cv::line(m, *prev, *it, cv::Scalar(50, 205, 50), 2);
+        prev = it;
+    }
+}
+
+void drawText(cv::Mat &m, const cv::RotatedRect &rc, const std::string &str) {
+    const int    fface   = cv::FONT_HERSHEY_SIMPLEX;
+    const double scale   = 0.7;
+    const int    thick   = 1;
+          int    base    = 0;
+    const auto text_size = cv::getTextSize(str, fface, scale, thick, &base);
+
+    std::vector<cv::Point2f> tmp_points(4);
+    rc.points(tmp_points.data());
+    const auto tl_point_idx = custom::OCVCropLabels::topLeftPointIdx(tmp_points);
+    cv::Point text_pos = tmp_points[tl_point_idx];
+    text_pos.x = std::max(0, text_pos.x);
+    text_pos.y = std::max(text_size.height, text_pos.y);
+
+    cv::rectangle(m,
+                  text_pos + cv::Point{0, base},
+                  text_pos + cv::Point{text_size.width, -text_size.height},
+                  CV_RGB(50, 205, 50),
+                  cv::FILLED);
+    const auto white = CV_RGB(255, 255, 255);
+    cv::putText(m, str, text_pos, fface, scale, white, thick, 8);
+}
+
+} // anonymous namespace
+} // namespace vis
+
+int main(int argc, char *argv[])
+{
+    try {
+    cv::CommandLineParser cmd(argc, argv, keys);
```

Comment:
You can avoid extra indentation from try block in this way:
```
static void run(int argc, char *argv[])
{
    ... code from try goes here ...
}

int main(int argc, char *argv[])
{
    try
    {
        run(argc, argv);
    }
    catch ...
    {
        ...
    }
    return 0;
}
```

Reply:
This try was only needed to debug Windows problems where exception text is not printed in console by default; so I removed it now.